packages/compiler-cli/linker/test/ast/typescript/typescript_ast_host_spec.ts

```diff
@@ -0,0 +1,274 @@
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+import {TypeScriptAstHost} from '../../../src/ast/typescript/typescript_ast_host';
+
+describe('TypeScriptAstHost', () => {
+  let host: TypeScriptAstHost;
+  beforeEach(() => host = new TypeScriptAstHost());
+
+  describe('getSymbolName()', () => {
+    it('should return the name of an identifier', () => {
+      expect(host.getSymbolName(expr(`someIdentifier`))).toEqual('someIdentifier');
+    });
+
+    it('should return the name of an identifier at the end of a property access chain', () => {
+      expect(host.getSymbolName(expr(`a.b.c.someIdentifier`))).toEqual('someIdentifier');
+    });
+
+    it('should return null if the expression has no identifier', () => {
+      expect(host.getSymbolName(expr(`42`))).toBe(null);
+    });
+  });
+
+  describe('isStringLiteral()', () => {
+    it('should return true if the expression is a string literal', () => {
+      expect(host.isStringLiteral(expr(`"moo"`))).toBe(true);
+      expect(host.isStringLiteral(expr(`'moo'`))).toBe(true);
+    });
+
+    it('should return false if the expression is not a string literal', () => {
+      expect(host.isStringLiteral(expr(`true`))).toBe(false);
+      expect(host.isStringLiteral(expr(`someIdentifier`))).toBe(false);
+      expect(host.isStringLiteral(expr(`42`))).toBe(false);
+      expect(host.isStringLiteral(rhs(`x = {}`))).toBe(false);
+      expect(host.isStringLiteral(expr(`[]`))).toBe(false);
+      expect(host.isStringLiteral(expr(`null`))).toBe(false);
+      expect(host.isStringLiteral(expr(`'a' + 'b'`))).toBe(false);
+    });
+
+    it('should return false if the expression is a template string', () => {
+      expect(host.isStringLiteral(expr(`\`moo\``))).toBe(false);
+    });
+  });
+
+  describe('parseStringLiteral()', () => {
+    it('should extract the string value', () => {
+      expect(host.parseStringLiteral(expr(`"moo"`))).toEqual('moo');
+      expect(host.parseStringLiteral(expr(`'moo'`))).toEqual('moo');
+    });
+  });
+
+  describe('isNumericLiteral()', () => {
+    it('should return true if the expression is a number literal', () => {
+      expect(host.isNumericLiteral(expr(`42`))).toBe(true);
+    });
+
+    it('should return false if the expression is not a number literal', () => {
+      expect(host.isStringLiteral(expr(`true`))).toBe(false);
+      expect(host.isNumericLiteral(expr(`"moo"`))).toBe(false);
+      expect(host.isNumericLiteral(expr(`'moo'`))).toBe(false);
+      expect(host.isNumericLiteral(expr(`someIdentifier`))).toBe(false);
+      expect(host.isNumericLiteral(rhs(`x = {}`))).toBe(false);
+      expect(host.isNumericLiteral(expr(`[]`))).toBe(false);
+      expect(host.isNumericLiteral(expr(`null`))).toBe(false);
+      expect(host.isNumericLiteral(expr(`'a' + 'b'`))).toBe(false);
+      expect(host.isNumericLiteral(expr(`\`moo\``))).toBe(false);
```

Comment:
No tagged template here so this could be:

```suggestion
      expect(host.isNumericLiteral(expr('`moo`'))).toBe(false);
```

In fact, many of the template strings in this file can be regular string literals.

Reply:
Yeah they are leftover from when all of these were tagged template literals.

Reply:
I have removed all the backticks